e2e_playwright/st_tabs_selection_test.py

```diff
@@ -87,9 +87,26 @@ def test_maintains_selection_when_same_name_exists(
     wait_for_app_run(app)
     # Add Tab 3
     control_buttons.nth(0).click()
+    wait_for_app_run(app)
```

Comment:
nit: maybe we can replace these button clicks with the `click_button` from app_utils which auto-waits for rerun and doesn't uses this ugly index-based access.

Reply:
Good idea, update the entire suite using the `click_button` API